src/aspire/abinitio/commonline_c3_c4.py

```diff
@@ -0,0 +1,363 @@
+
+        epsilon = 1e-2
+        max_iters = 100
+
+        # Initialize candidate eigenvectors
+        vec = np.random.randn(n_vijs)
+        vec = vec / norm(vec)
+        dd = 1
+        itr = 0
+
+        # Power method iterations
+        while itr < max_iters and dd > epsilon:
+            itr += 1
+            vec_new = self._signs_times_v(vijs, vec)
+            # vec_new, eigenvalues = qr(vec_new)
+            vec_new = vec_new / norm(vec_new)
+            dd = norm(vec_new - vec)
+            vec = vec_new
+
+        logger.info(
+            f"Power method used {itr} iterations. Maximum iterations set to {max_iters}."
+        )
+
+        # We need only the signs of the eigenvector
+        J_sync = np.sign(vec)
+
+        return J_sync
+
+    def _signs_times_v(self, vijs, vec):
+        """
+        For each triplet of outer products vij, vjk, and vik, the associated elements of the "signs"
+        matrix are populated with +1 or -1 and multiplied by the corresponding elements of
+        the current candidate eigenvector supplied by the power method. The new candidate eigenvector
+        is updated for each triplet.
+
+        :param vijs: Nchoose2 x 3 x 3 array, where each 3x3 slice holds the outer product of vi and vj.
+
+        :param vec: The current candidate eigenvector of length Nchoose2 from the power method.
+
+        :return: New candidate eigenvector of length Nchoose2. The product of the signs matrix and vec.
+        """
+        # All pairs (i,j) and triplets (i,j,k) where i<j<k
+        indices = np.arange(self.n_ims)
+        pairs = [(i, j) for idx, i in enumerate(indices) for j in indices[idx + 1 :]]
+        trips = [
+            (i, j, k)
+            for idx, i in enumerate(indices)
+            for j in indices[idx + 1 :]
+            for k in indices[j + 1 :]
+        ]
+
+        # There are four possible signs configurations for each triplet of nodes vij, vik, vjk.
+        signs = np.zeros((4, 3))
+        signs[0] = [1, 1, 1]
+        signs[1] = [-1, 1, -1]
+        signs[2] = [-1, -1, 1]
+        signs[3] = [1, -1, -1]
+
+        J = np.diag((-1, -1, 1))
+        v = vijs
+        new_vec = np.zeros_like(vec)
+
+        for _, (i, j, k) in enumerate(trips):
+            ij = pairs.index((i, j))
+            jk = pairs.index((j, k))
+            ik = pairs.index((i, k))
+
+            # Conditions for relative handedness. The minimum of these conditions determines
+            # the relative handedness of the triplet of vijs.
+            c = np.zeros(4)
+            c[0] = norm(v[ij] @ v[jk] - v[ik])
+            c[1] = norm(J @ v[ij] @ J @ v[jk] - v[ik])
+            c[2] = norm(v[ij] @ J @ v[jk] @ J - v[ik])
+            c[3] = norm(v[ij] @ v[jk] - J @ v[ik] @ J)
+
+            min_c = np.argmin(c)
+
+            # Assign signs +-1 to edges between nodes vij, vik, vjk.
+            s_ij_jk = signs[min_c][0]
+            s_ik_jk = signs[min_c][1]
+            s_ij_ik = signs[min_c][2]
```

Comment:
does `s_ij_jk, s_ik_jk, s_ij_ik = signs[min_c]` work?

Reply:
It does. Thanks. done.